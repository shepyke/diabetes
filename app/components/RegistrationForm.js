/**
 * Diabetes
 * https://github.com/shepyke/diabetes
 * @flow
 */

import React, { Component } from 'react';
import {
    Platform,
    Text,
    View,
    TextInput,
    KeyboardAvoidingView,
    TouchableOpacity,
    AsyncStorage,
} from 'react-native';
import Styles from "../config/Styles";
import DatePicker from 'react-native-datepicker';
import RadioForm,
    {
        RadioButton,
        RadioButtonInput,
        RadioButtonLabel
    } from 'react-native-simple-radio-button';

var gender_props = [
    {label: 'Female    ', value: 'F' },
    {label: 'Male', value: 'M' }
];

export default class RegistrationForm extends Component<{}> {

    constructor(props){
        super(props);
        this.state = {
            user: {
                username: '',
                password: '',
                repassword: '',
                firstName: '',
                lastName: '',
                email: '',
                profileImage: '',
                birthDate: '',
                gender: '',
                type: '',
                bloodSugarAVG: '',
            }
        }
    }

    submit = async() => {
       try {
            fetch('http://172.20.10.12:3000/users/registration', {
                method: 'POST',
                headers: {
                    'Accept': 'application/json',
                    'Content-Type': 'application/json',
                },
                body: JSON.stringify({
                    user:{
                        firstName: this.state.user.firstName,
                        lastName: this.state.user.lastName,
                        username: this.state.user.username,
                        email: this.state.user.email,
                        birthDate: this.state.user.birthDate,
                        //profileImage: this.state.user.profileImage,
                        gender: this.state.user.gender,
                        type: this.state.user.type,
                        //bloodSugarAVG: this.state.user.bloodSugarAVG,
                        password: this.state.user.password,
                        repassword: this.state.user.repassword,
                    }
                })
            })
            .then((response) => response.json())
            .then ((res) => {
                if(res.success === true){
                    console.log("hopp itt a response: " + res.user);
                    AsyncStorage.setItem('user', JSON.stringify(res.user));
                    this.props.navigation.navigate('Profile');
                    alert('You have successfully registered');
                }else{
                    alert(res.message);
                }
            })
            .done();
        }catch(err){
           console.log(err);
       }

    }

    goToLogin = () => {
        this.props.navigation.navigate('Home');
    }

    render() {
        return (
            <View style={Styles.wrapper}>
                <KeyboardAvoidingView id='registration' behavior='padding' />
                <View style={Styles.container}>

                    <Text style={Styles.header}>Registration</Text>

                    <View style={Styles.rowContainer}>

                        <TextInput
                            style={[Styles.textInput, {width: '45%'}]}
                            placeholder='First Name'
                            onChangeText={
                                (firstName) => {
                                    const user = Object.assign({},
                                        this.state.user, { firstName: firstName });
                                    this.setState({ user: user });
                                }}
                            underlineColorAndroid='transparent'
                            placeholderTextColor='#718792'
                        />

                        <TextInput
                            style={[Styles.textInput, {width: '45%', marginLeft: '7%'}]}
                            placeholder='Last name'
                            onChangeText={
                                (lastName) => {
                                    const user = Object.assign({},
                                        this.state.user, { lastName: lastName });
                                    this.setState({ user: user });
                                }}
                            underlineColorAndroid='transparent'
                            placeholderTextColor='#718792'
                        />
                    </View>

                    <RadioForm
                        radio_props={gender_props}
                        labelColor={'#718792'}
                        initial={'F'}
                        formHorizontal={true}
                        labelHorizontal={true}
                        buttonColor={'#718792'}
                        animation={true}
                        onPress={(gender) => {
                            const user = Object.assign({},
                                this.state.user, { gender: gender });
                            this.setState({ user: user })
                        }}
                    />

                    <TextInput
                        style={Styles.textInput}
                        placeholder='Email'
                        onChangeText={
                            (email) => {
                                const user = Object.assign({},
                                    this.state.user, { email: email });
                                this.setState({ user: user });
                            }}
                        underlineColorAndroid='transparent'
                        placeholderTextColor='#718792'
                    />

                    <TextInput
                        style={Styles.textInput}
                        placeholder='Username'
                        onChangeText={
                            (username) => {
                                const user = Object.assign({},
                                    this.state.user, { username: username });
                                this.setState({ user: user });
                            }}
                        underlineColorAndroid='transparent'
                        placeholderTextColor='#718792'
                    />

                    <DatePicker
                        style={Styles.birthDate}
                        date={this.state.user.birthDate}
                        mode="date"
                        placeholder="Select your birthday"
                        format="YYYY-MM-DD"
                        minDate="1900-01-01"
                        maxDate={new Date()}
                        confirmBtnText="Confirm"
                        cancelBtnText="Cancel"
                        showIcon={false}
                        customStyles={{
                            dateInput: {
                                borderWidth: 0,
                                position: 'absolute',
                                marginLeft: 0,
                                left: 0,
                            },
                            dateText: {
                                color: '#fff',
                            },
                            placeholderText: {
                                color: '#718792',
                            },
                        }}
                        onDateChange={(birthDate) => {
                            const user = Object.assign({},
                                this.state.user, { birthDate: birthDate });
                            this.setState({ user: user });
                        }}
                    />
                    <View style={Styles.rowContainer}>
                        <TextInput
                            style={[Styles.textInput, {width: '45%'}]}
                            placeholder='Password'
                            placeholderTextColor='#718792'
                            onChangeText={
                                (password) => {
                                    const user = Object.assign({},
                                        this.state.user, { password: password });
                                    this.setState({ user: user });
                                }}
                            underlineColorAndroid='transparent'
                            secureTextEntry={true}
                        />

                        <TextInput
                            style={[Styles.textInput, {width: '45%', marginLeft: '7%'}]}
                            placeholder='Password again'
                            placeholderTextColor='#718792'
                            onChangeText={
                                (repassword) => {
                                    const user = Object.assign({},
                                        this.state.user, { repassword: repassword });
                                    this.setState({ user: user });
                                }}
                            underlineColorAndroid='transparent'
                            secureTextEntry={true}
                        />
                    </View>

                    <TouchableOpacity
                        style={Styles.button}
                        onPress={this.submit}>
                        <Text style={Styles.text}>Sign up</Text>
                    </TouchableOpacity>

                    <Text style={Styles.bottomtext}>
                        <Text>Already have account? </Text>
                        <Text
                            style={Styles.register}
                            onPress={this.goToLogin}>
                            Go to the Login page!
                        </Text>
                    </Text>

                </View>
            </View>
        );
    }
}