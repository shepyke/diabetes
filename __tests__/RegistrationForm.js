import 'react-native';
import React from 'react';
import RegistrationForm from '../app/components/RegistrationForm';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

//Start of testId: regist-1
it('renders the normal TextInput component', () => {
    const TextInput = require('TextInput');
    const tree = renderer
        .create(<TextInput value="username" />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
//End of testId: regist-1

//Start of testId: regist-2
it('renders the secure TextInput component for passwords', () => {
    const TextInput = require('TextInput');
    const tree = renderer
        .create(<TextInput secureTextEntry={true} value="passwordnotshown" />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
//End of testId: regist-2

//Start of testId: regist-3
test('Passwords during user registration', () => {
    let user = {username: 'testUser', password: 'testPassword', repassword: 'testpassword'};

    expect(user.password).not.toMatch(user.repassword);
})
//End of testId: regist-3

