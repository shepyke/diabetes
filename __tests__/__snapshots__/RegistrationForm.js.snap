// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the normal TextInput component 1`] = `
<TextInput
  allowFontScaling={true}
  value="username"
/>
`;

exports[`renders the secure TextInput component for passwords 1`] = `
<TextInput
  allowFontScaling={true}
  secureTextEntry={true}
  value="passwordnotshown"
/>
`;
